### PR TITLE
Geolocation front end

### PR DESCRIPTION
**Pull request by WantingLi816**
image
show closest bus stop when clicking on the button on the top right corner. Only makes sense when a route is selected from the drop down

### CODE REVIEW

**Comment by WantingLi816** [comment]
made another change to the backend so that only closest bus stop with given route number will be returned. otherwise return 404. This change should make the feature more useful.

**Review by dennis-fan — APPROVED** [review]
thank you lisa!

**Comment by dburenok** [comment]
Great work guys

**Review by dburenok — APPROVED** [review]

File: back-end/controllers/busStopController.js [hunk 1]
```diff
@@ -12,19 +13,30 @@ const findStopLatLong = async (req, res) => {
...
+
   const stop = await BusStopLocation.findOne({
     geoloc: {
       $near: {
         $geometry: {
            type: "Point" ,
            coordinates: [ long , lat ]
         },
-        $maxDistance: 500,
+        $maxDistance: 50,
```
Comment (dburenok) [inline back-end/controllers/busStopController.js:32]: ~~caught a bug~~ nvm - for some reason the endpoint is not returning the nearest bus stop even thought the exact same DB query does return a number of them, looking into it

**Pull request merged**